text/3663-feature-deps.md

```diff
@@ -0,0 +1,112 @@
+- Feature Name: feature-deps
+- Start Date: 2024-06-20
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC extends Cargo's structured table form of features to include a `deps`
+key, to unambiguously specify packages the feature depends on without having to
+use the `dep:foo` microformat.
+
+```toml
+[features]
+myfeature = { deps = ["some-crate", "another-crate"] }
+# This is equivalent to `myfeature = ["dep:some-crate", "dep:another-crate"]`
+# This can also be written in shorthand form:
+myfeature.deps = ["some-crate", "another-crate"]
+```
+
+This depends on
+[RFC 3416 (feature metadata)](https://github.com/rust-lang/rfcs/pull/3416),
+which defines the structured table form of features.
+
+# Motivation
+[motivation]: #motivation
+
+Cargo features can depend on other features, on other packages, and on features
+of other packages. [RFC 3143](https://github.com/rust-lang/rfcs/pull/3143) and
+[RFC 3491](https://github.com/rust-lang/rfcs/pull/3491) have migrated Cargo to
+distinguishing a feature's different kinds of dependencies with different
+syntax: `dep:some-crate` for a crate, `some-crate?/feature` to enable an
+optional dependency's feature without force-enabling the optional dependency
+itself, and `some-crate/feature` as a shorthand combining the two.
+
+This has created somewhat of a microformat within the dependencies list, and
+this microformat adds complexity compared to other parts of the Cargo manifest,
+which distinguish different kinds of things via different keys or sections. (By
+way of example, we have `dev-dependencies` and `build-dependencies`, rather
+than a microformat like `dependencies."build:xyz"`.)
+
+Now that [RFC 3416](https://github.com/rust-lang/rfcs/pull/3416) has defined a
+structured table form for defining features, this RFC takes a first step
+towards providing structured equivalents in place of the microformat.
```

Comment:
We also have the `cfg` microformat.

Overall, I feel like judicious use of microformats extends what a standardized grammar allows and can enhance the format.  For me, this motivation feels very weak and the change possibly a net negative while encouraging a wide ecosystem churn that I'm not seeing the pay off for.